server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeRole.java

```diff
@@ -20,7 +18,7 @@
 /**
  * Represents a node role.
  */
-public abstract class DiscoveryNodeRole implements Comparable<DiscoveryNodeRole> {
+public class DiscoveryNodeRole implements Comparable<DiscoveryNodeRole> {
```

Comment:
This could be a followup, but I wonder if we could make this final now. Seems like the only need for override is the enabled by default predicate, but this could be a lambda passed to the second constructor?

Reply:
Yeah, I will take a close look at opportunities for things like this when I get around to removing the plugin extension point for node roles.